woocommerce/woocommerce-android
# TPP: Tracking

Closes: #8146


### Description

The PR adds the following events:
* `MANAGE_CARD_READERS_AUTOMATIC_DISCONNECT_BUILT_IN_READER` (happens when a user is connected to a built-in reader and the goes to the "manage card reader" menu - we automatically disconnect from built-in reader in this case)
* `CARD_PRESENT_SELECT_READER_TYPE_BUILT_IN_TAPPED` (on card reader type selection dialog click on TPP option)
* `CARD_PRESENT_SELECT_READER_TYPE_BLUETOOTH_TAPPED` (on card reader type selection dialog click on bluetooth option)
* `CARD_PRESENT_TAP_TO_PAY_NOT_AVAILABLE` (if TPP is not available on the device then we track it every time when a user initiates IPP flow). It contains a property with key = `reason` and one of the values:
  * `TapToPayDisabled`
  * `SystemVersionNotSupported`
  * `GooglePlayServicesNotAvailable`
  * `NfcNotAvailable`
  * `CountryNotSupported`

---
@joshheald On android, we track screen show events with a generic one, e.g.:
```
Tracked: view_shown, Properties: {"name":"CardReaderTypeSelectionDialogFragment","is_debug":true}
```

So I think we don't need to add `card_present_select_reader_type_shown` event? Wdyt?

Also, you may consider adding `CARD_PRESENT_TAP_TO_PAY_NOT_AVAILABLE` if you think it's a need of that on iOS

---

### Testing instructions

* Use logcat with filter word: "Tracked"
* Go via the flow as described in the PR description
* Notice all 4 newly added events there, for instance

```
Tracked: card_present_tap_to_pay_not_available, Properties: {"reason":"NfcNotAvailable","plugin_slug":"woocommerce-payments","country":"US","blog_id":192152755,"is_wpcom_store":true,"is_debug":true}
Tracked: view_shown, Properties: {"name":"CardReaderTypeSelectionDialogFragment","is_debug":true}
Tracked: card_present_select_reader_type_built_in_tapped, Properties: {"plugin_slug":"woocommerce-payments","country":"US","blog_id":192152755,"is_wpcom_store":true,"is_debug":true}
Tracked: card_present_select_reader_type_bluetooth_tapped, Properties: {"plugin_slug":"woocommerce-payments","country":"US","currency":"USD","card_reader_model":"COTS_DEVICE","blog_id":192152755,"is_wpcom_store":true,"is_debug":true}
Tracked: manage_card_readers_automatic_disconnect_built_in_reader, Properties: {"plugin_slug":"woocommerce-payments","country":"US","currency":"USD","card_reader_model":"COTS_DEVICE","blog_id":192152755,"is_wpcom_store":true,"is_debug":true}
```
